### ISSUE TITLE
No error reported for overload ambiguity with extension methods with identical signatures and `params` arrays

### ISSUE THREAD

**Issue by cston** [dotnet/roslyn#72696]
**Version Used**: 

main: 7ecf61f6d02b5605c97da1408e0440548f6bf863

**Steps to Reproduce**:

Compile the following:
```csharp
using System.Collections;
using System.Collections.Generic;

class MyCollection<T> : IEnumerable<T>
{
    IEnumerator<T> IEnumerable<T>.GetEnumerator() => throw null;
    IEnumerator IEnumerable.GetEnumerator() => throw null;
}

static class ExtensionsA
{
    public static void Add<T>(this MyCollection<T> collection, params string[] args) { }
}

static class ExtensionsB
{
    public static void Add<T>(this MyCollection<T> collection, params string[] args) { }
}

class Program
{
    static void Main()
    {
        var x = new MyCollection<object>();
        x.Add("");
        
        var y = new MyCollection<object> { "" };
    }
}
```

**Expected Behavior**:

The following errors are expected, as reported by 4.10.0-3.24161.9 (b0be4981).
```
(25,11): error CS0121: The call is ambiguous between the following methods or properties:
    'ExtensionsA.Add<T>(MyCollection<T>, params string[])' and
    'ExtensionsB.Add<T>(MyCollection<T>, params string[])'
(27,44): error CS0121: The call is ambiguous between the following methods or properties:
    'ExtensionsA.Add<T>(MyCollection<T>, params string[])' and
    'ExtensionsB.Add<T>(MyCollection<T>, params string[])'
```

**Actual Behavior**:

No errors.

**Comment by cston** [comment]
cc @AlekseyTs

**Comment by AlekseyTs** [comment]
It looks like ```IsBetterCollectionExpressionConversion``` is not designed to be called with identical type, which happens for `params` scenario from ```IsBetterParamsCollectionType```.
